llvm/test/tools/llvm-symbolizer/approximate-line-handcrafted.s

```diff
@@ -0,0 +1,489 @@
+#	.byte 0                                      # reference to dir0\
+#	.byte 0                                      # modification time\
+#	.byte 0                                      # length of file (unavailable)\
+#	.byte 0                                      # end of filenames\
+#.Lprologue_end:\
+#	.byte 0x05, 66                               # DW_LNS_set_column (80)\
+#	.byte 0x0a                                   # DW_LNS_set_prologue_end\
+#	.byte 0x00, 9, 2                             # DW_LNS_set_address\
+#	.quad 0x30a0                                 # Address Value (0x00000000000030a0)\
+#	.byte 0x01                                   # DW_LNS_copy\
+#	.byte 0x02                                   # DW_LNS_advance_pc\
+#	.uleb128 0x06                                # (addr +=6, op-index +=0)\
+#	.byte 0, 1, 1                                # DW_LNE_end_sequence\
+#	.byte 0x05, 68                               # DW_LNS_set_column (68)\
+#	.byte 0x0a                                   # DW_LNS_set_prologue_end\
+#	.byte 0x00, 9, 2                             # DW_LNE_set_address\
+#	.quad 0x1000                                 # Address Value (0x0000000000001000)\
+#	.byte 0x11                                   # (address += 0,  line += -1,  op-index += 0)\
+#	.byte 0x05, 39                               # DW_LNS_set_column (39)\
+#	.byte 0x06                                   # DW_LNS_negate_stmt\
+#	.byte 0x84                                   # (address += 8,  line += 2,  op-index += 0)\
+#	.byte 0x05, 68                               # DW_LNS_set_column (68)\
+#	.byte 0x06                                   # DW_LNS_negate_stmt\
+#	.byte 0x0a                                   # DW_LNS_set_prologue_end\
+#	.byte 0x83                                   # (address += 8,  line += 1,  op-index += 0)\
+#	.byte 0x06                                   # DW_LNS_negate_stmt\
+#	.byte 0x2b                                   # (address += 2,  line += -3,  op-index += 0)\
+#	.byte 0x5b                                   # (address += 5,  line += 3,  op-index += 0)\
+#	.byte 0x05, 39                               # DW_LNS_set_column (39)\
+#	.byte 0x2e                                   # (address += 2,  line += 0,  op-index += 0)\
+#	.byte 0x04, 2                                # DW_LNS_set_file (2)\
+#	.byte 0x05, 1                                # DW_LNS_set_column (1)\
+#	.byte 0x06                                   # DW_LNS_negate_stmt\
+#	.byte 0x0a                                   # DW_LNS_set_prologue_end\
+#	.byte 0x76                                   # (address += 7,  line += 2,  op-index += 0)\
+#	.byte 0x02                                   # DW_LNS_advance_pc\
+#	.uleb128 0x06                                # (addr += 6, op-index += 0)\
+#	.byte 0, 1, 1                                # DW_LNE_end_sequence\
+#	.Lunit_end:' main.s
+
+#sed -n p main.s
+
+#--- main.s
+	.section	.debug_loc,"",@progbits
```

Comment:
Can you symbolize with just the line table, and not loc, info, etc? It certainly seems like a lot of other debug info (like a loclist seems entirely unneeded for any symbolizing test, but there's also just a lot of DIEs in the debug_info, etc)

Reply:
Current revision, has removed the debug_loc sections(by shortening the source code). Manually deleting the debug_* sections is not an option here since then as I commented we get weird behaviour in llvm-symbolizer output because of missing other debug_* sections.